### ISSUE TITLE
feRedirect doesn't include the section parameter in the url 

### ISSUE THREAD

**Issue by agoat** [contao/core-bundle#673]
When opening an article from the backend article list, the generated url doesn't include the section like the readmore link generated by [\Contao\ModuleArticle](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/modules/ModuleArticle.php#L115).

The redirectToFrontendPage method in [\Contao\Controller](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/Controller.php#L1287-L1315) simply adding the article alias (or id). 
This should extended (on line 1296-1299) to something like this:
```php
if ($varArticle !== null)
{
	$objArticle = \ArticleModel::findPublishedByIdOrAliasAndPid($varArticle, $intPage)
	
	if ($objArticle !== null)
	{
		$varArticle = '/articles/' . (($objArticle->inColumn != 'main') ? $objArticle->inColumn . ':' : '') . $varArticle;
	}
}
```

**Comment by leofeyer** [comment]
I'm pretty sure we have changed this on purpose.

**Comment by agoat** [comment]
But the current behavior leads to a broken frontend layout, if an article, that exist in a layout section other than the main column, is opened from the backend directly.

**Comment by leofeyer** [comment]
The code is exactly the same in Contao 3.5 and in Contao 4:

* [Controller.php in Contao 3.5](https://github.com/contao/core/blob/master/system/modules/core/library/Contao/Controller.php#L1311-L1339)
* [Controller.php in Contao 4.3](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/library/Contao/Controller.php#L1287-L1315)

**Comment by agoat** [comment]
IMO it's then wrong in both versions.

If an article, that resides in an section other than main, is opened from the backend, this article will then also be rendered from the article module in the main section.

This happens only in the backend, so it's a minor bug, but it is a bug.

**Comment by asaage** [comment]
The behaviour not optimal.
In case you display an article as teaser on Page feRedirect points you to ```/page/articles/article.html``` wich is fine. If you don't display it as teaser it does the same but the correct url imo would be ```/page.html``` then.

**Comment by leofeyer** [comment]
We have dropped the article preview some time ago, so now you are always redirected to the page:

https://github.com/contao/core/blob/master/system/modules/core/controllers/BackendMain.php#L270

> In case you display an article as teaser on page - feRedirect points you to ```/page/articles/article.html``` 

?

**Comment by agoat** [comment]
We are talking about clicking on the icons in front of the article title.

![article_feredirect](https://cloud.githubusercontent.com/assets/1402434/22106300/62c7876c-de48-11e6-9255-2f6b572d232e.jpg)

**Comment by asaage** [comment]
![image](https://cloud.githubusercontent.com/assets/4138336/22106292/50e16f9a-de48-11e6-9347-48c60ea58fd7.png)

![image](https://cloud.githubusercontent.com/assets/4138336/22106351/a29538b2-de48-11e6-962c-900084c1c278.png)

**Comment by leofeyer** [comment]
> Maybe just remove ```.'&amp;article='.($row['alias'] ?: $row['id']).'``` and show us the page.

Would that be ok for you, too, @agoat?

**Comment by agoat** [comment]
And what about extending the redirectToFrontendPage method in the Controller class?

```php
if ($varArticle !== null)
{
	$objArticle = \ArticleModel::findPublishedByIdOrAliasAndPid($varArticle, $intPage)
	
	if ($objArticle !== null)
	{
		$varArticle = '/articles/' . (($objArticle->inColumn != 'main') ? $objArticle->inColumn . ':' : '') . $varArticle;
	}
	else
	{
		$varArticle = ''
	}
}
```

I like the possibility to open an article directly from the backend. IMO we shouldn't remove that.

**Comment by asaage** [comment]
Messing around in the demo i noticed that the section can be part of the url. I did not know about that before.
That's funny an weired...
https://demo.contao.org/en/text-elements.html
https://demo.contao.org/en/text-elements/articles/text-elements.html
https://demo.contao.org/en/text-elements/articles/left:text-elements.html
https://demo.contao.org/en/text-elements/articles/header:text-elements.html
https://demo.contao.org/en/text-elements/articles/main:text-elements.html
https://demo.contao.org/en/text-elements/articles/nonsense:text-elements.html

**Comment by leofeyer** [comment]
Ok, please replace the `Controller::redirectToFrontendPage()` method with the following code:

```php
	/**
	 * Redirect to a front end page
	 *
	 * @param integer $intPage    The page ID
	 * @param string  $strArticle An optional article alias
	 * @param boolean $blnReturn  If true, return the URL and don't redirect
	 *
	 * @return string The URL of the target page
	 */
	protected function redirectToFrontendPage($intPage, $strArticle=null, $blnReturn=false)
	{
		if (($intPage = intval($intPage)) <= 0)
		{
			return '';
		}

		$objPage = \PageModel::findWithDetails($intPage);

		if ($objPage === null)
		{
			return '';
		}

		$strParams = null;

		// Add the /article/ fragment (see #673)
		if ($strArticle !== null && ($objArticle = \ArticleModel::findByAlias($strArticle)) !== null)
		{
			$strParams = '/articles/' . (($objArticle->inColumn != 'main') ? $objArticle->inColumn . ':' : '') . $strArticle;
		}

		$strUrl = $objPage->getFrontendUrl($strParams);

		// Make sure the URL is absolute (see #4332)
		if (strncmp($strUrl, 'http://', 7) !== 0 && strncmp($strUrl, 'https://', 8) !== 0)
		{
			$strUrl = \Environment::get('base') . $strUrl;
		}

		if (!$blnReturn)
		{
			$this->redirect($strUrl);
		}

		return $strUrl;
	}
```

Does it fix the problem?

**Comment by agoat** [comment]
Works!

And looks more elegant than my suggestion ;-)

**Comment by leofeyer** [comment]
Fixed in c7875feef7213be39c0abf425408cf185dd4fb2d.
